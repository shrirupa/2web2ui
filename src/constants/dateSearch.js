export const RELATIVE_DATE_OPTIONS = [
  'hour',
  'day',
  '7days',
  '10days',
  'custom'
];
