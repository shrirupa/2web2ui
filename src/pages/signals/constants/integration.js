export const batchStatusOptions = [
  { label: 'Success', value: 'success' },
  { label: 'Validation', value: 'validation' },
  { label: 'System', value: 'system' },
  { label: 'Decompression', value: 'decompress' },
  { label: 'Duplicates', value: 'duplicate_batch' },
  { label: 'Empty batches', value: 'empty_batch' }
];
