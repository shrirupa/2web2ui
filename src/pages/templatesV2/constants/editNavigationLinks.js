import EditContents from '../components/EditContents';
import TemplateSettings from '../components/settings/EditSettings';
import EditContentsPrimaryArea from '../components/EditContentsPrimaryArea';

const editNavigationLinks = [
  {
    content: 'Content',
    key: 'content',
    render: EditContents,
    renderPrimaryArea: EditContentsPrimaryArea,
    routeKey: 'content'
  },
  {
    content: 'Settings',
    key: 'settings',
    render: TemplateSettings,
    renderPrimaryArea: EditContentsPrimaryArea,
    routeKey: 'settings'
  }
];

export default editNavigationLinks;
