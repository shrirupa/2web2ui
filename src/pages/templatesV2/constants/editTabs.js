import EditAmpSection from '../components/EditAmpSection';
import EditHtmlSection from '../components/EditHtmlSection';
import EditTextSection from '../components/EditTextSection';
import TestDataSection from '../components/TestDataSection';

const editTabs = [
  {
    content: 'HTML',
    key: 'html',
    render: EditHtmlSection
  },
  {
    content: 'AMP HTML',
    key: 'amp_html',
    render: EditAmpSection
  },
  {
    content: 'Text',
    key: 'text',
    render: EditTextSection
  },
  {
    content: 'Test Data',
    key: 'test_data',
    render: TestDataSection
  }
];

export default editTabs;
