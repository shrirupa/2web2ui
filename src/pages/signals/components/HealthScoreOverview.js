/* eslint-disable max-lines */
import _ from 'lodash';
import React from 'react';
import { Panel, Tooltip } from '@sparkpost/matchbox';
import { InfoOutline } from '@sparkpost/matchbox-icons';
import SummaryTable, { Column } from 'src/components/summaryTable';
import { setSubaccountQuery } from 'src/helpers/subaccounts';
import { HEALTH_SCORE_INFO } from '../constants/info';
import { DEFAULT_VIEW } from '../constants/summaryTables';
import FacetDataCell from './dataCells/FacetDataCell';
import NumericDataCell from './dataCells/NumericDataCell';
import SparklineDataCell from './dataCells/SparklineDataCell';
import WoWDataCell from './dataCells/WoWDataCell';
import WoWHeaderCell from './dataCells/WoWHeaderCell';
import styles from './SpamTrapOverview.module.scss';

class HealthScoreOverview extends React.Component {

  componentDidMount() {
    this.resetTable();
  }

  // assumptions, signalOptions and summaryTable should never both change on the same update and
  // resetting signal options will trigger a summary table reset which calls getData
  componentDidUpdate(prevProps) {
    const { signalOptions, summaryTable } = this.props;

    if (prevProps.signalOptions !== signalOptions) {
      this.resetTable();
    }

    if (prevProps.summaryTable !== summaryTable) {
      this.getData();
    }
  }

  resetTable = () => {
    const { facet, resetSummaryTable, tableName } = this.props;
    let options;

    if (facet.key === 'sid') {
      options = DEFAULT_VIEW;
    }

    resetSummaryTable(tableName, options);
  }

  getData = () => {
    const { getHealthScore, signalOptions, summaryTable } = this.props;
    let { subaccount } = signalOptions;
    let order;
    let orderBy;

    if (summaryTable.order) {
      order = summaryTable.order.ascending ? 'asc' : 'desc';
      orderBy = summaryTable.order.dataKey;
    }

    if (subaccount && subaccount.id === undefined) {
      subaccount = undefined; // unset
    }

    getHealthScore({
      facet: signalOptions.facet,
      filter: signalOptions.facetSearchTerm,
      from: signalOptions.from,
      limit: summaryTable.perPage,
      offset: (summaryTable.currentPage - 1) * summaryTable.perPage,
      order,
      orderBy,
      relativeRange: signalOptions.relativeRange,
      subaccount,
      to: signalOptions.to
    });
  }

  handleClick = (facetId, subaccountId) => ({ date }) => {
    const { facet, history } = this.props;
    let search;

    if (facet.key === 'sid' && facetId === -1) {
      return; // ignore
    }

    if (subaccountId >= 0) {
      search = setSubaccountQuery(subaccountId);
    }

    history.push({
      pathname: `/signals/health-score/${facet.key}/${facetId}`,
      search,
      state: {
        date
      }
    });
  }

  render() {
    const {
      data, error, facet, loading, signalOptions, subaccounts, tableName, totalCount, hideTitle, header
    } = this.props;

    const subaccountFilter = _.get(signalOptions, 'subaccount.id');
    const isCustomRange = signalOptions.relativeRange === 'custom';

    const noFacetSelected = facet.key === 'sid';
    const noSubaccountFilter = subaccountFilter === undefined;

    return (
      <Panel>
        {/*
          This prop is only because v1 requires this title, but v2 does not.
          Should be removed once v2 is released.
        */}
        {!hideTitle && (
          <div className={styles.Header}>
            <h5>Health Score Summary</h5>
            <div className={styles.Tooltip}>
              <Tooltip
                children={<InfoOutline className={styles.TooltipIcon} size={18} />}
                content={HEALTH_SCORE_INFO}
                dark
                horizontalOffset="-1rem"
                right
              />
            </div>
          </div>
        )}
        {header && <Panel.Section>{header}</Panel.Section>}
        <SummaryTable
          data={data}
          empty={data.length === 0}
          error={error && error.message}
          loading={loading}
          tableName={tableName}
          totalCount={totalCount}
        >
          {(noSubaccountFilter || noFacetSelected) && (
            <Column
              dataKey="sid"
              label="Subaccount"
              sortable
              width={noFacetSelected ? '30%' : '15%'}
              component={({ sid }) => (
                <FacetDataCell
                  dimension="health-score"
                  facet="sid"
                  id={sid}
                  name={_.get(_.find(subaccounts, { id: sid }), 'name')}
                  truncate
                />
              )}
            />
          )}
          {!noFacetSelected && (
            <Column
              dataKey={facet.key}
              label={facet.label}
              sortable
              width={noSubaccountFilter ? '15%' : '30%'}
              component={(props) => (
                <FacetDataCell
                  dimension="health-score"
                  facet={facet.key}
                  id={props[facet.key]}
                  subaccountId={props.sid}
                  truncate
                />
              )}
            />
          )}
          <Column
            dataKey="history"
            label="Daily Health Score"
            width='30%'
            component={({ history, ...data }) => {
              const id = data[facet.key];

              return (
                <SparklineDataCell
                  data={history}
                  dataKey="health_score"
                  label="Health Score"
                  onClick={this.handleClick(id, data.sid)}
                  relative={false}
                />
              );
            }}
          />
          <Column
            align="right"
            dataKey="current_health_score"
            label={isCustomRange ? 'Score' : 'Current Score'}
            sortable
            width="12.5%"
            component={({ current_health_score }) => (
              <NumericDataCell value={current_health_score} />
            )}
          />
          <Column
            align="right"
            dataKey="WoW"
            label={<WoWHeaderCell/>}
            width="12.5%"
            component={({ WoW }) => (
              <WoWDataCell value={WoW} />
            )}
          />
          <Column
            align="right"
            dataKey="average_health_score"
            label="Average Score"
            width="15%"
            component={({ average_health_score }) => (
              <NumericDataCell value={average_health_score} />
            )}
          />
        </SummaryTable>
      </Panel>
    );
  }
}

export default HealthScoreOverview;
