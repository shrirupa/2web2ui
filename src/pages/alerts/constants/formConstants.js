import React, { Fragment } from 'react';
import { EmailIcon, SlackIcon, WebhookIcon } from 'src/components/icons';
import { emails, ifStringPresent, maxItems, noDuplicateItems } from 'src/helpers/validation';
import { UnstyledLink } from '@sparkpost/matchbox';
export const FORM_NAME = 'alertForm';

export const METRICS = {
  monthly_sending_limit: 'Monthly Sending Limit',
  health_score: 'Health Score',
  block_bounce_rate: 'Block Bounce Rate',
  hard_bounce_rate: 'Hard Bounce Rate',
  soft_bounce_rate: 'Soft Bounce Rate',
  injection_count: 'Injection Count',
  blacklist: 'Blacklists',
};

export const FILTERS_FRIENDLY_NAMES = {
  none: 'None',
  ip_pool: 'IP Pool',
  mailbox_provider: 'Mailbox Provider',
  sending_domain: 'Sending Domain',
  sending_ip: 'Sending IP',
  subaccount_id: 'Subaccounts',
  blacklist_provider: 'Blacklists',
  blacklist_resource: 'Domains or IPs',
};

export const RECOMMENDED_METRIC_VALUE = {
  monthly_sending_limit: {
    raw: {
      gt: 80,
    },
  },
  health_score: {
    raw: {
      lt: 80,
      gt: 70,
    },
    week_over_week: {
      gt: 10,
    },
    day_over_day: {
      gt: 10,
    },
  },
  block_bounce_rate: {
    raw: {
      gt: 20,
    },
  },
  hard_bounce_rate: {
    raw: {
      gt: 20,
    },
  },
  soft_bounce_rate: {
    raw: {
      gt: 20,
    },
  },
};

const webhookSubtitle = (
  <Fragment>
    {'Create a webhook for this alert. '}
    <UnstyledLink
      external
      to={'https://www.sparkpost.com/docs/user-guide/alerts/#webhook-payload-sample'}
    >
      Sample payload
    </UnstyledLink>
  </Fragment>
);

const slackSubtitle = (
  <Fragment>
    {'Integrate this alert with Slack. '}
    <UnstyledLink external to={'https://api.slack.com/incoming-webhooks'}>
      How to create a Slack Incoming webhook
    </UnstyledLink>
  </Fragment>
);

export const NOTIFICATION_CHANNEL_DATA = {
  emails: {
    icon: <EmailIcon />,
    subtitle: 'Receive notifications through email. One address per line. Up to 10 addresses.',
    fieldProps: {
      validate: [
        ifStringPresent(emails),
        ifStringPresent(maxItems(10)),
        ifStringPresent(noDuplicateItems),
      ],
      placeholder: 'example@email.com',
      multiline: true,
    },
  },
  slack: {
    icon: <SlackIcon />,
    subtitle: slackSubtitle,
    fieldProps: {
      placeholder: 'https://hooks.slack.com/services/T00/B00/XX ',
    },
  },
  webhook: {
    icon: <WebhookIcon />,
    subtitle: webhookSubtitle,
    fieldProps: {
      placeholder: 'https://example.com/webhook-target',
    },
  },
};

export const SIGNALS_FILTERS = ['none', 'ip_pool', 'mailbox_provider', 'sending_domain'];

export const REALTIME_FILTERS = ['sending_ip', 'mailbox_provider', 'sending_domain'];

export const SOURCE_FRIENDLY_NAMES = {
  raw: 'Score',
  week_over_week: 'Week over Week',
  day_over_day: 'Day over Day',
};

export const OPERATOR_FRIENDLY_NAMES = {
  gt: 'Above',
  lt: 'Below',
};

export const NOTIFICATION_CHANNELS = ['emails', 'slack', 'webhook'];

export const DEFAULT_FORM_VALUES = {
  name: '',
  metric: '',
  subaccounts: [],
  sending_ip: [],
  mailbox_provider: [],
  sending_domain: [],
  single_filter: { filter_type: 'none', filter_values: [] },
  source: 'raw',
  operator: 'gt',
  value: 0,
  muted: false,
};
