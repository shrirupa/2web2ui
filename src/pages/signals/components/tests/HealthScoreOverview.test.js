import React from 'react';
import { shallow } from 'enzyme';
import SummaryTable from 'src/components/summaryTable';
import { DEFAULT_VIEW } from '../../constants/summaryTables';
import HealthScoreOverview from '../HealthScoreOverview';

describe('HealthScoreOverview', () => {
  const subject = (props = {}) => shallow(
    <HealthScoreOverview
      data={[
        {
          current_health_score: 98,
          domain: 'example.com',
          history: [
            { date: '2018-01-13', health_score: 98 }
          ],
          average_health_score: 98,
          WoW: 0.1,
          sid: 123
        }
      ]}
      facet={{
        key: 'domain',
        label: 'Domain'
      }}
      getHealthScore={() => {}}
      loading={false}
      resetSummaryTable={() => {}}
      signalOptions={{
        facet: 'domain',
        facetSearchTerm: 'example.com',
        from: '2015-01-01',
        relativeRange: '14days',
        subaccount: {
          id: 123
        },
        to: '2015-01-05'
      }}
      subaccounts={{
        123: { id: 123, name: 'Test Subaccount' }
      }}
      summaryTable={{
        currentPage: 1,
        order: { ascending: true, dataKey: 'domain' },
        perPage: 10
      }}
      tableName="Test"
      totalCount={1}
      {...props}
    />
  );

  it('renders overview panel with controls and table', () => {
    expect(subject()).toMatchSnapshot();
  });

  it('renders empty summary table', () => {
    const wrapper = subject({ data: []});
    expect(wrapper.find(SummaryTable).prop('empty')).toEqual(true);
  });

  it('renders error message', () => {
    const wrapper = subject({
      data: [],
      error: new Error('Oh no!')
    });

    expect(wrapper.find(SummaryTable).prop('error')).toEqual('Oh no!');
  });

  it('renders loader', () => {
    const wrapper = subject({
      data: [],
      loading: true
    });

    expect(wrapper.find(SummaryTable).prop('loading')).toEqual(true);
  });

  it('renders both subaccount and facet columns', () => {
    const wrapper = subject({
      signalOptions: {
        id: undefined,
        name: 'Master & All Subaccounts'
      }
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('renders custom date range', () => {
    const wrapper = subject({ signalOptions: { relativeRange: 'custom' }});
    expect(wrapper.find('Column[dataKey="current_health_score"]').prop('label')).toEqual('Score');
  });

  it('does not render title', () => {
    const wrapper = subject({ hideTitle: true });
    expect(wrapper.find('div[className="Header"]')).not.toExist();
  });

  it('requests reset on mount', () => {
    const resetSummaryTable = jest.fn();
    subject({ resetSummaryTable });
    expect(resetSummaryTable).toHaveBeenCalledWith('Test', undefined);
  });

  it('requests reset to default view for subaccount view on mount', () => {
    const resetSummaryTable = jest.fn();
    subject({ facet: { key: 'sid', label: 'Subaccounts' }, resetSummaryTable });
    expect(resetSummaryTable).toHaveBeenCalledWith('Test', DEFAULT_VIEW);
  });

  it('requests table reset on signal options update', () => {
    const resetSummaryTable = jest.fn();
    const wrapper = subject();
    wrapper.setProps({ resetSummaryTable, signalOptions: {}});

    expect(resetSummaryTable).toHaveBeenCalledWith('Test', undefined);
  });

  it('requests data on summary table update', () => {
    const getHealthScore = jest.fn();
    const summaryTable = {
      currentPage: 2,
      perPage: 10
    };
    const wrapper = subject();

    wrapper.setProps({ getHealthScore, summaryTable });

    expect(getHealthScore).toHaveBeenCalledWith({
      facet: 'domain',
      filter: 'example.com',
      from: '2015-01-01',
      limit: 10,
      offset: 10,
      order: undefined,
      orderBy: undefined,
      relativeRange: '14days',
      subaccount: {
        id: 123
      },
      to: '2015-01-05'
    });
  });

  it('requests ordered data on summary table update', () => {
    const getHealthScore = jest.fn();
    const summaryTable = {
      currentPage: 1,
      order: { ascending: true, dataKey: 'domain' },
      perPage: 10
    };
    const wrapper = subject();

    wrapper.setProps({ getHealthScore, summaryTable });

    expect(getHealthScore).toHaveBeenCalledWith(expect.objectContaining({
      order: 'asc',
      orderBy: 'domain'
    }));
  });

  it('requests data without subaccount data on summary table update', () => {
    const getHealthScore = jest.fn();
    const signalOptions = {
      facet: 'domain',
      facetSearchTerm: 'example.com',
      relativeRange: '14days',
      subaccount: {
        id: undefined,
        name: 'Master & All Subaccounts'
      }
    };
    const summaryTable = {
      currentPage: 2,
      perPage: 10
    };
    const wrapper = subject();

    wrapper.setProps({ getHealthScore, signalOptions, summaryTable });

    expect(getHealthScore).toHaveBeenCalledWith(expect.objectContaining({
      subaccount: undefined
    }));
  });

  describe('history component', () => {
    const factory = (pageProps) => (props) => {
      const wrapper = subject(pageProps);
      const Column = wrapper.find('Column[dataKey="history"]').prop('component');

      return shallow(<Column domain="example.com" sid={123} {...props} />);
    };

    it('redirects to details page when bar is clicked', () => {
      const historyPush = jest.fn();
      const wrapper = factory({ history: { push: historyPush }})();
      wrapper.simulate('click', { date: '2018-01-13' });

      expect(historyPush).toHaveBeenCalledWith({
        pathname: '/signals/health-score/domain/example.com',
        search: '?subaccount=123',
        state: { date: '2018-01-13' }
      });
    });

    it('redirects to details page when dot is clicked', () => {
      const historyPush = jest.fn();
      const wrapper = factory({ history: { push: historyPush }})();
      wrapper.simulate('click', { date: '2018-01-13' });

      expect(historyPush).toHaveBeenCalledWith({
        pathname: '/signals/health-score/domain/example.com',
        search: '?subaccount=123',
        state: { date: '2018-01-13' }
      });
    });

    it('ignores master and all subaccount clickthrough to details page', () => {
      const historyPush = jest.fn();
      const wrapper = factory({
        facet: {
          key: 'sid',
          label: 'Subaccount'
        },
        history: { push: historyPush }
      })({
        sid: -1
      });
      wrapper.simulate('click', { date: '2018-01-13' });

      expect(historyPush).not.toHaveBeenCalled();
    });
  });

  describe('current score column component', () => {
    it('renders current score', () => {
      const wrapper = subject();
      const Column = wrapper.find('Column[dataKey="current_health_score"]').prop('component');
      const columnWrapper = shallow(<Column current_health_score={99} />);

      expect(columnWrapper).toMatchSnapshot();
    });
  });
});
