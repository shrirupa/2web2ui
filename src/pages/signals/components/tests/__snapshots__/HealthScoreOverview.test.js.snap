// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HealthScoreOverview current score column component renders current score 1`] = `
<NumericDataCell
  value={99}
/>
`;

exports[`HealthScoreOverview renders both subaccount and facet columns 1`] = `
<Panel>
  <div
    className="Header"
  >
    <h5>
      Health Score Summary
    </h5>
    <div
      className="Tooltip"
    >
      <Tooltip
        bottom={true}
        content="
  This is a predictive score that monitors your email health to identify problems before
  they negatively impact email delivery. This score is informed by message events,
  including bounces, spam trap hits, and engagement across our entire network.
"
        dark={true}
        forcePosition={false}
        horizontalOffset="-1rem"
        preferredDirection={
          Object {
            "bottom": true,
            "left": false,
            "right": true,
            "top": false,
          }
        }
        right={true}
      >
        <InfoOutline
          className="TooltipIcon"
          size={18}
        />
      </Tooltip>
    </div>
  </div>
  <Connect(SummaryTable)
    data={
      Array [
        Object {
          "WoW": 0.1,
          "average_health_score": 98,
          "current_health_score": 98,
          "domain": "example.com",
          "history": Array [
            Object {
              "date": "2018-01-13",
              "health_score": 98,
            },
          ],
          "sid": 123,
        },
      ]
    }
    empty={false}
    loading={false}
    tableName="Test"
    totalCount={1}
  >
    <Column
      component={[Function]}
      dataKey="sid"
      label="Subaccount"
      sortable={true}
      width="15%"
    />
    <Column
      component={[Function]}
      dataKey="domain"
      label="Domain"
      sortable={true}
      width="15%"
    />
    <Column
      component={[Function]}
      dataKey="history"
      label="Daily Health Score"
      width="30%"
    />
    <Column
      align="right"
      component={[Function]}
      dataKey="current_health_score"
      label="Current Score"
      sortable={true}
      width="12.5%"
    />
    <Column
      align="right"
      component={[Function]}
      dataKey="WoW"
      label={<WoWHeaderCell />}
      width="12.5%"
    />
    <Column
      align="right"
      component={[Function]}
      dataKey="average_health_score"
      label="Average Score"
      width="15%"
    />
  </Connect(SummaryTable)>
</Panel>
`;

exports[`HealthScoreOverview renders overview panel with controls and table 1`] = `
<Panel>
  <div
    className="Header"
  >
    <h5>
      Health Score Summary
    </h5>
    <div
      className="Tooltip"
    >
      <Tooltip
        bottom={true}
        content="
  This is a predictive score that monitors your email health to identify problems before
  they negatively impact email delivery. This score is informed by message events,
  including bounces, spam trap hits, and engagement across our entire network.
"
        dark={true}
        forcePosition={false}
        horizontalOffset="-1rem"
        preferredDirection={
          Object {
            "bottom": true,
            "left": false,
            "right": true,
            "top": false,
          }
        }
        right={true}
      >
        <InfoOutline
          className="TooltipIcon"
          size={18}
        />
      </Tooltip>
    </div>
  </div>
  <Connect(SummaryTable)
    data={
      Array [
        Object {
          "WoW": 0.1,
          "average_health_score": 98,
          "current_health_score": 98,
          "domain": "example.com",
          "history": Array [
            Object {
              "date": "2018-01-13",
              "health_score": 98,
            },
          ],
          "sid": 123,
        },
      ]
    }
    empty={false}
    loading={false}
    tableName="Test"
    totalCount={1}
  >
    <Column
      component={[Function]}
      dataKey="domain"
      label="Domain"
      sortable={true}
      width="30%"
    />
    <Column
      component={[Function]}
      dataKey="history"
      label="Daily Health Score"
      width="30%"
    />
    <Column
      align="right"
      component={[Function]}
      dataKey="current_health_score"
      label="Current Score"
      sortable={true}
      width="12.5%"
    />
    <Column
      align="right"
      component={[Function]}
      dataKey="WoW"
      label={<WoWHeaderCell />}
      width="12.5%"
    />
    <Column
      align="right"
      component={[Function]}
      dataKey="average_health_score"
      label="Average Score"
      width="15%"
    />
  </Connect(SummaryTable)>
</Panel>
`;
